CONTRIBUTING.md

```diff
@@ -21,3 +315,72 @@ Use [this guide][style] to maintain consistent style across the Conjur project.
 [style]: STYLE.md
 [tests]: README.md#Testing
 [issues]: https://github.com/cyberark/conjur/issues
+
+## Changelog maintenance
+
+The [changelog file](CHANGELOG.md) is maintained based on
+[Keep a Changelog](http://keepachangelog.com/en/1.0.0/) guidelines.
+
+Each accepted change to the Conjur code (documentation and website updates
+excepted) requires adding a changelog entry to the corresponding `Added`,
+`Changed`, `Deprecated`, `Removed`, `Fixed` and/or `Security` sub-section (add
+one as necessary) of the _Unreleased_ section in the changelog.
+
+Bumping the version number after each and every change is not required,
+advised nor expected. Valid reasons to bump the version are for example:
+
+- enough changes have accumulated,
+- an important feature has been implemented,
+- an external project depends on one of the recent changes.
+
+## Releasing
+
+### Verify and update dependencies
+1. Review the [NOTICES.txt](./NOTICES.txt) file and ensure it reflects the current
+   set of dependencies in the [Gemfile](./Gemfile)
+1. If a new dependency has been added, a dependency has been dropped, or a version
+   has changed since the last tag - make sure the NOTICES file is up-to-date with
+   the new versions
+
+### Update the version and changelog
+1. Examine the changelog and decide on the version bump rank (major, minor, patch).
+1. Change the title of _Unreleased_ section of the changelog to the target
+version.
+  - Be sure to add the date (ISO 8601 format) to the section header.
+1. Add a new, empty _Unreleased_ section to the changelog.
+  - Remember to update the references at the bottom of the document.
+1. Change VERSION file to reflect the change. This file is used by some scripts.
+1. Commit these changes (including the changes to NOTICES.txt, if there are any).
+   `Bump version to x.y.z` is an acceptable commit message.
+1. Push your changes to a branch, and get the PR reviewed and merged.
+
+### Tag the version
+1. Tag the version on the master branch using eg. `git tag -s v0.1.1`. Note this
+  requires you to be able to sign releases. Consult the
+  [github documentation on signing commits](https://help.github.com/articles/signing-commits-with-gpg/)
+  on how to set this up.
+  - git will ask you to enter the tag message. These will become the release notes.
+  Format should be like this (note the subject line and message):
+
+        Version x.y.z
+
+        This is a human-readable overview of the changes in x.y.z. It should be a
+        consise, at-a-glance summary. It certainly isn't a direct copy-and-paste
+        from the changelog.
+
+1. Push the tag: `git push vx.y.z` (or `git push origin vx.y.z` if you are working
```

Comment:
What do you think about using `git push v1.2.3` or something like that.  putting the `v` (which follows our convention) next to the `x` reads a bit weird to me.

Reply:
yeah, just noticed there was a v0.1.1 in the example above it too - updated to make them all consistent